### ISSUE TITLE
[Menu] MenuButton doesn't toggle menu on Safari

### ISSUE THREAD

**Issue by laurensnl** [chakra-ui/chakra-ui#587]
**Describe the bug**
On Safari, clicking the MenuButton when the Menu is open hides the Menu and immediately re-opens it.

**Expected Behavior**
I would expect the menu to close after clicking the MenuButton for the second time, like in Chrome.

**To reproduce**
1. Go to https://chakra-ui.com/menu in Safari.
2. Press any menu button to open the menu
3. Press the button again

**Desktop (please complete the following information):**
 - OS: Mac OS X 10.15.4
 - Browser Safari
 - Version 13.1

**Comment by DanLeCornu** [comment]
Can confirm, doesn't work on Firefox for me either

**Comment by navin-moorthy** [comment]
I am not sure, how this will affect the accessibility,

But `onMouseDown` - `event.preventDefault()` code on MenuButton will fix the issue,

![image](https://user-images.githubusercontent.com/39694575/79033499-54888780-7bcc-11ea-8bfa-8ce13b27c6e2.png)

Because in Safari and Firefox, the `onBlur` of MenuList fires after `onMouseDown`.

Example: https://codepen.io/mudassir0909/pen/eIHqB

I can open a PR if my solution is valid.

**Comment by segunadebayo** [comment]
Hi guys,

Thanks for this issue and for suggesting a walkaround @navin-moorthy. I think that should do that trick, for now.

We've solved this permanently in the preparation for the next release.

**Comment by laurensnl** [comment]
Thanks @navin-moorthy & @segunadebayo!

**Comment by laurensnl** [comment]
@segunadebayo Issue still exists on v0.8.0 when using Safari

**Comment by with-heart** [comment]
@laurensnl Does the workaround still work for this issue? This will be fixed in the 1.0 release, which is coming asap.

**Comment by laurensnl** [comment]
> @laurensnl Does the workaround still work for this issue in 0.8.0? This will be fixed in the 1.0 release (which I think is what @segunadebayo meant by "next release" rather than 0.8.0), which is coming asap.

The workaround still works. Just wanted to let you know that the issue is still present in the latest release.

**Comment by with-heart** [comment]
@laurensnl Gotcha! Ty!

**Comment by gunnartorfis** [comment]
I'm experiencing the same issue on Safari, works on Chrome. I'm using "@chakra-ui/react": "1.1.2".

I wanted to use Popover but I experienced [this issue](https://github.com/chakra-ui/chakra-ui/issues/3238) with that one.

**Comment by laurensnl** [comment]
@with-heart @segunadebayo Just wanted to let you know that this issue still seems to be happening on the PopoverTrigger component. The `onMouseDown={(e) => e.preventDefault()}` workaround works.
